src/services/data/data.service.spec.ts

```diff
@@ -1,27 +1,16 @@
 import { DataService } from './data.service';
-import { mockMeasureName } from '../../utils/mocks.spec';
-import { Duration } from 'luxon';
+import { PreferenceService } from '../preference/preference.service';
 
 describe('DataService', () => {
```

Comment:
Could you please add tests for the DataService functionality (i.e. mock the preferenceService providing dataset$)? Can be done in later PR if needed.

Reply:
Oh, I should have done that. I will add these tests in this PR in the old way, and then refactor them when resolving #54 .

Reply:
Part of the tests has been added, while others may need some mock datasets or refactoring. The remaining tests should be added when resolving #54